torso-bundle.js

```diff
@@ -256,149 +369,61 @@
-      var i, newTag, currTag,
-        newElem, currElem,
-        newChildren, currChildren,
-        newAttributes, currentAttributes,
-        replacementDOM, attrNode,
-        skip, ignoreIdx, childForceRefresh,
-        hardRefreshes = [],
-        ignoreElementsLen = ignoreElements ? ignoreElements.length : 0;
-
-      // Handle tagname changes with full replacement
-      newTag = newDOM.prop('tagName');
-      currTag = currentDOM.prop('tagName');
-      if (newTag !== currTag) {
-        replacementDOM = $('<' + newTag + '>' + newDOM.html() + '</' + newTag + '>');
-        currentDOM.replaceWith(replacementDOM);
-        currentDOM = replacementDOM;
-      }
-
-      // Attribute removing old values
-      newAttributes = newDOM.get(0).attributes;
-      currentAttributes = currentDOM.get(0).attributes;
-      while (currentAttributes.length > 0) {
-        currentAttributes.removeNamedItem(currentAttributes[0].name);
-      }
-
-      // Attribute setting for new values
-      _.each(newAttributes, function(attrib) {
-        attrNode = document.createAttribute(attrib.name);
-        attrNode.value = attrib.value;
-        currentAttributes.setNamedItem(attrNode);
-      });
-
-      // Quick check if we need to bother comparing sub-levels
-      if (currentDOM.html() === newDOM.html()) {
-        return false;
-      }
-
-      newChildren = newDOM.children();
-      currChildren = currentDOM.children();
-
-      // If the DOM lists are different sizes, perform a hard refresh
-      if (newChildren.length !== currChildren.length) {
-        currentDOM.html(newDOM.html());
-        return true;
-      }
-
-      // Compare and set content if this is a leaf node
-      if (currChildren.length === 0) {
-        currentDOM.html(newDOM.html());
-        return false;
-      }
-
-      // Perform a recursive hotswap for all children elements
-      for (i = 0; i < currChildren.length; i++) {
-        skip = false;
-        newElem = $(newChildren[i]);
-        currElem = $(currChildren[i]);
-        if (ignoreElements) {
-          for (ignoreIdx = 0; ignoreIdx < ignoreElementsLen; ignoreIdx++) {
-            if (currElem.is(ignoreElements[ignoreIdx])) {
-              skip = true;
-              break;
-            }
-          }
-        }
-        if (!skip) {
-          childForceRefresh = this.hotswap(currElem, newElem, ignoreElements, returnRefreshTree);
-          if (returnRefreshTree) {
-            hardRefreshes.push(childForceRefresh);
-          }
-        }
-      }
-      return hardRefreshes;
-    },
+    // See above function declaration for method-level documentation
+    hotswap: hotswap,
 
     /**
      * Produces a copy of the element tag with attributes but with no contents
-     * @param el {jQuery element} the element to be copied
-     * @return a shallow copy of the element with no children but with attributes
+     * @param el {Element} the DOM element to be copied
```

Comment:
I'm changing the interface for the copyTopElement method to take and return plain Element objects instead of jQuery objects. This is why I made a same change to ListView